src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -57,26 +64,75 @@ public static void main(String[] args) {
+          helpMessage = "To use one of your macros, simply write \"@MacroBot MacroName <your message> \".";
         }
         break;
+
       case "MESSAGE":
-        // Sends request to execution module
-        String email = event.at("/message/sender/email").asText();
         String message = event.at("/message/text").asText();
+        String[] words = message.split(" ");
         String threadId = event.at("/message/thread/name").asText();
-        replyText = macroExecutionModule.execute(email, message, threadId);
+
+        // Check that a message was written.
+        if (words.length <= 1) {
+            replyText = "You must type a message when you message me. Please type \"@MacroBot help\" for more instructions.";
+            break;
+        }
+        // Reply to the help message.
+        if (words[1].equalsIgnoreCase("help")) {  
+            replyText = helpMessage; 
+        }        
+        else if (words.length >= 2) {
+            // If a macro is being initialized for a room.
+            if (words[1].equalsIgnoreCase("initiate")) {
+                // getMacroName() requires the macroName to be the second word of the message, so remove the word "initiate". This will put the macro name as the second word in the message.
+                List<String> wordsWithoutInitiate = new LinkedList(Arrays.asList(words));
+                wordsWithoutInitiate.remove("initiate");
+                String newMessage = String.join(" ", wordsWithoutInitiate);
+                String macroName = MacroExecutionModuleImplementation.getMacroName(newMessage, threadId);
+
+                macroCreatorEmail = event.at("/message/sender/email").asText();
+                Map<String, String> macroToCreator = new HashMap<String, String> ();
+                macroToCreator.put(macroName, macroCreatorEmail);
+
+                roomToMacro.put(event.at("/space/name").asText(), macroToCreator);
+                replyText = "The " + macroName + " macro belonging to " + macroCreatorEmail + " has been initiated for this room. All users in this room can use this macro.";
+            }
+            // If someone is trying to use the macro
+            else if (roomToMacro.containsKey(event.at("/space/name").asText())) {
+                String macroName = MacroExecutionModuleImplementation.getMacroName(message, threadId);
+                Map<String, String> macroToCreator = roomToMacro.get(event.at("/space/name").asText());
+                macroCreatorEmail = macroToCreator.get(macroName);
+                String userEmail = event.at("/message/sender/email").asText();
+                replyText = macroExecutionModule.execute(userEmail, macroCreatorEmail, message, threadId, macroName);
+            }  
+            else {
+                replyText = "You do not have any macros initialized for this room.";
+            } 
+        }
+        else {
+            replyText = "That is not a valid message.";
+        }
```

Comment:
I like this! I think it would be good to separate it into a different function starting at line 86? What do you think

Reply:
I agree. Would you keep this function in Bot.java or put it in the execution class? @tsa87 any thoughts on this?

Reply:
Good idea! Would it make sense if the ExecutionModule dealt with the initialization with `Map<string<Map<string, string>>` and let `Bot.java` only handle getting and sending messages to hangout?